Tasks/Common/PowershellHelpers/Helpers.ps1

```diff
@@ -75,6 +75,29 @@ function Invoke-ActionWithRetries {
     Trace-VstsLeavingInvocation $MyInvocation
 }
 
+function Get-TempDirectoryPath
+{
+    <#
+    .SYNOPSIS
+    Returns a temp directory path. Uses Agent.TempDirectory if available if shorter than env temp
+    #>
+
+    Param ()
+
+    $agentVersion = Get-VstsTaskVariable -Name 'agent.version'
+    $envTemp = $env:Temp
+    if ($agentVersion -and (([version]'2.115.0').CompareTo([version]$agentVersion) -lt 1))
```

Comment:
Consider the ifs can be combined using:
$envTemp = $env:TEMP
$agentTemp = $env:AGENT_TEMPDIRECTORY

if ($agentTemp -and $agentTemp.Length -lt $tmp.Length)
{
    return $agentTemp
}
return $envTemp

If the agent doesn't support AGENT_TEMPDIRECTORY it will not be defined

Reply:
I have removed check for agent version. Still using Get-VstsTaskVariable to get agent temp dir rather than depend on $env